tools/generate-code.py

```diff
@@ -83,8 +85,8 @@ def generate_fromwire_field(field, allfields, lang):
     if is_array:
         if lang == 'js':
             print('    v = [];\n'
-                  '    for (let i = 0; {limit}; i++) {{\n'
-                  '        v.push(fromwire_{ftype}(buffer));\n'
+                  '    for (let i = 0; i<{limit}; i++) {{\n'
+                  '        v.push(fromwire_{ftype}(Buffer.from(buffer[i].toString(16),"hex")));\n'
```

Comment:
I also fixed this one: if you look further up, limitstr is set to:

1. 'i < {}'.format(field.fieldtype.arraysize)
2. 'i < lenfield_{}'.format(field.fieldtype.lenfield.name)

So they must all be of the same form (if we had more tests you would have found this).  So either change the other ones, or leave the "i <" in limitstr.